projects/angular-auth-oidc-client/src/lib/iframe/check-session.service.ts

```diff
@@ -49,17 +48,21 @@ export class CheckSessionService implements OnDestroy {
 
   ngOnDestroy(): void {
     this.stop();
-    this.document.defaultView.removeEventListener(
-      'message',
-      this.iframeMessageEventListener,
-      false
-    );
+    const windowAsDefaultView = this.document.defaultView;
```

Comment:
better to consider that we have a deal with CSR.
Otherwise, we always have to validate and check whether .Window is defined or not. 
I suggest you provide a dedicated token or forbid access here and validate in the initial state of the lib.